authentication/GetsWithRedirectThenPost.js

```diff
@@ -0,0 +1,157 @@
+
+    // Post the authentication
+    msg = doPost(helper, paramsValues, credentials);
+    return msg;
+}
+
+
+function doGet(url, helper) {
+    //decode URI. Useful when there are encoded parameters in the URI
+    var requestUri = new URI(decodeURIComponent(url), false);
+    var requestMethod = HttpRequestHeader.GET;
+
+    // Build the GET request header
+    var requestHeader = new HttpRequestHeader(requestMethod, requestUri, HttpHeader.HTTP11);
+
+    // Build the GET request message
+    var msg = helper.prepareMessage();
+
+    msg.setRequestHeader(requestHeader);
+    msg.getRequestHeader().setContentLength(msg.getRequestBody().length());
+
+    // Send the GET request message
+    doLog("Sending " + requestMethod + " request to " + requestUri);
+    // sendAndReceive without following redirects
+    // this allows us to manually add the redirect request/responses in ZAP history, making it much easier to understand what is going on. 
+    // With followRedirect=true, all request/responses and their cookies are aggregated in a single line in ZAP history tab. 
+    helper.sendAndReceive(msg, false);
+    doLog("Received response status code: " + msg.getResponseHeader().getStatusCode());
+    return msg;
+}
+
+
+
+function doPost(helper, paramsValues, credentials) {
+    // Prepare the login submission request details
+    var requestUri = new URI(paramsValues.get("Submission Form URL"), false);
+    var requestMethod = HttpRequestHeader.POST;
+
+    // Build the submission request body using the credential values
+    var requestBody = paramsValues.get("Username field") + "=" + encodeURIComponent(credentials.getParam("usernameField"));
+    requestBody += "&" + paramsValues.get("Password field") + "=" + encodeURIComponent(credentials.getParam("passwordField"));
+
+    // Build the submission request header
+    var requestHeader = new HttpRequestHeader(requestMethod, requestUri, HttpHeader.HTTP11);
+
+    // Build the submission request message
+    var msg = helper.prepareMessage();
+    msg.setRequestHeader(requestHeader);
+    msg.setRequestBody(requestBody);
+    msg.getRequestHeader().setContentLength(msg.getRequestBody().length());
+
+    // Send the submission request message
+    doLog("Sending " + requestMethod + " request to " + requestUri + " with body: " + requestBody);
+    // In case of redirect on the POST with the cookie not being set into the state, see TwoStepAuthentication.js
+    helper.sendAndReceive(msg, true);
+    doLog("Received response status code: " + msg.getResponseHeader().getStatusCode());
+    return msg;
+}
+
+
+
+// This function is called during the script loading to obtain a list of the names of the required configuration parameters,
+// that will be shown in the Session Properties -  Authentication panel for configuration. They can be used
+// to input dynamic data into the script, from the user interface (e.g. a login URL, name of POST parameters etc.)
+
```

Comment:
Remove extra blank line here and before the comment?

Reply:
done